### PR TITLE
Stable HLO runtime tests

### PR DESCRIPTION
**Pull request by mmanzoorTT**
Requires #1164

### CODE REVIEW

**Comment by ddilbazTT** [comment]
concat tests fail because it does not support f64. We can replace f64 with f32. The following worked for me:

```
// REQUIRES: stablehlo
// RUN: rm -rf %t.ttnn
// RUN: ttmlir-opt --stablehlo-to-ttir-pipeline %s | \
// RUN:     ttmlir-opt --ttir-to-ttnn-backend-pipeline="system-desc-path=%system_desc_path%" | \
// RUN:     ttmlir-translate --ttnn-to-flatbuffer > %t.ttnn

module @jit_concat attributes {} {
  func.func public @test_concat(%arg0: tensor<32x32xf32>, %arg1: tensor<32x64xf32>) -> tensor<32x96xf32> {
    %0 = "stablehlo.concatenate"(%arg0, %arg1) {
    dimension = 1 : i64
    } : (tensor<32x32xf32>, tensor<32x64xf32>) -> tensor<32x96xf32>
    return %0 : tensor<32x96xf32>
  }
  func.func public @test_concat_2(%arg0: tensor<3x2xi64>, %arg1: tensor<1x2xi64>) -> tensor<4x2xi64> {
    %0 = "stablehlo.concatenate"(%arg0, %arg1) {
    dimension = 0 : i64
    } : (tensor<3x2xi64>, tensor<1x2xi64>) -> tensor<4x2xi64>
    return %0 : tensor<4x2xi64>
  }
  func.func public @test_concat_3(%arg0: tensor<4x3xf32>, %arg1: tensor<4x5xf32>) -> tensor<4x8xf32> {
    %0 = "stablehlo.concatenate"(%arg0, %arg1) {
    dimension = 1 : i64
    } : (tensor<4x3xf32>, tensor<4x5xf32>) -> tensor<4x8xf32>
    return %0 : tensor<4x8xf32>
  }
  func.func public @test_concat_4(%arg0: tensor<128x64xf32>, %arg1: tensor<128x96xf32>) -> tensor<128x160xf32> {
    %0 = "stablehlo.concatenate"(%arg0, %arg1) {
      dimension = 1 : i64
    } : (tensor<128x64xf32>, tensor<128x96xf32>) -> tensor<128x160xf32>
    return %0 : tensor<128x160xf32>
  }
   func.func public @test_concat_5(%arg0: tensor<256x512xi64>, %arg1: tensor<256x256xi64>) -> tensor<256x768xi64> {
    %0 = "stablehlo.concatenate"(%arg0, %arg1) {
      dimension = 1 : i64
    } : (tensor<256x512xi64>, tensor<256x256xi64>) -> tensor<256x768xi64>
    return %0 : tensor<256x768xi64>
  }
   func.func public @test_concat_6(%arg0: tensor<64x32xi64>, %arg1: tensor<64x64xi64>) -> tensor<64x96xi64> {
    %0 = "stablehlo.concatenate"(%arg0, %arg1) {
      dimension = 1 : i64
    } : (tensor<64x32xi64>, tensor<64x64xi64>) -> tensor<64x96xi64>
    return %0 : tensor<64x96xi64>
  }
  func.func public @test_concat_7(%arg0: tensor<1000x128xi32>, %arg1: tensor<500x128xi32>) -> tensor<1500x128xi32> {
    %0 = "stablehlo.concatenate"(%arg0, %arg1) {
      dimension = 0 : i64
    } : (tensor<1000x128xi32>, tensor<500x128xi32>) -> tensor<1500x128xi32>
    return %0 : tensor<1500x128xi32>
  }
  func.func public @test_concat_8(%arg0: tensor<3x2x4x5xf32>, %arg1: tensor<3x2x4x3xf32>) -> tensor<3x2x4x8xf32> {
    %0 = "stablehlo.concatenate"(%arg0, %arg1) {
      dimension = 3 : i64
    } : (tensor<3x2x4x5xf32>, tensor<3x2x4x3xf32>) -> tensor<3x2x4x8xf32>
    return %0 : tensor<3x2x4x8xf32>
  }
  func.func public @test_concat_9(%arg0: tensor<8x4x6xi32>, %arg1: tensor<8x4x2xi32>) -> tensor<8x4x8xi32> {
    %0 = "stablehlo.concatenate"(%arg0, %arg1) {
      dimension = 2 : i64
    } : (tensor<8x4x6xi32>, tensor<8x4x2xi32>) -> tensor<8x4x8xi32>
    return %0 : tensor<8x4x8xi32>
  }
}
```

File: test/ttmlir/Silicon/StableHLO/unary/cbrt_op.mlir
Comment (ddilbazTT) [inline test/ttmlir/Silicon/StableHLO/unary/cbrt_op.mlir:1]: We should not use f64, we could use f32 instead. And we are missing x's. i.e. tensor<64x128f64> should be tensor<64x128xf64>

Comment (ddilbazTT) [reply test/ttmlir/Silicon/StableHLO/unary/cbrt_op.mlir:1]: This worked for me:

```
// REQUIRES: stablehlo
// RUN: rm -rf %t.ttnn
// RUN: ttmlir-opt --stablehlo-to-ttir-pipeline %s | \
// RUN:     ttmlir-opt --ttir-to-ttnn-backend-pipeline="system-desc-path=%system_desc_path%" | \
// RUN:     ttmlir-translate --ttnn-to-flatbuffer > %t.ttnn

#any_device = #tt.operand_constraint<dram|l1|scalar|tile|any_device|any_device_tile>
module @jit_eltwise_rsqrt attributes {} {
  func.func public @test_cbrt(%arg0: tensor<64x128xf32>) -> tensor<64x128xf32> {
    %0 = stablehlo.cbrt %arg0 : tensor<64x128xf32>
    return %0 : tensor<64x128xf32>
  }
}
```

**Review by kmitrovicTT — APPROVED** [review]
Should we have FileCheck directives in silicon tests or not?

**Comment by mmanzoorTT** [comment]
> Should we have FileCheck directives in silicon tests or not?

@kmitrovicTT I'll add `FileCheck` as per slack discussion.

**Comment by kmitrovicTT** [comment]
Keep this in mind https://tenstorrent.slack.com/archives/C07A30HRX7C/p1727698795548319

File: test/ttmlir/Silicon/StableHLO/maxpool2d_op.mlir [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+func.func public @test_maxpool2d(%arg0: tensor<1x128x128x32xbf16>) -> tensor<1x64x64x32xbf16> {
+  %0 = stablehlo.constant dense<0xFF80> : tensor<bf16>
+  %1 = stablehlo.broadcast_in_dim %0, dims = [] : (tensor<bf16>) -> tensor<bf16>
+  %2 = "stablehlo.reduce_window"(%arg0, %1) <{padding = dense<[[0, 0], [1, 1], [1, 1], [0, 0]]> : tensor<4x2xi64>, window_dimensions = array<i64: 1, 3, 3, 1>, window_strides = array<i64: 1, 2, 2, 1>}> ({
+  ^bb0(%arg2: tensor<bf16>, %arg3: tensor<bf16>):
+    %3 = stablehlo.maximum %arg2, %arg3 : tensor<bf16>
+    stablehlo.return %3 : tensor<bf16>
+  }) : (tensor<1x128x128x32xbf16>, tensor<bf16>) -> tensor<1x64x64x32xbf16>
+  return %2 : tensor<1x64x64x32xbf16>
+}
```
Comment (kmitrovicTT) [inline test/ttmlir/Silicon/StableHLO/maxpool2d_op.mlir:19]: Missing filecheck.

Comment (kmitrovicTT) [reply test/ttmlir/Silicon/StableHLO/maxpool2d_op.mlir:19]: Also, I am not sure about filecheck format you used. I am confident it passes FileCheck, but I don't find it as verbose as 

```mlir
  func.func public @test_remainder(%arg0: tensor<32x32xf32>, %arg1: tensor<32x32xf32>) -> tensor<32x32xf32> {
    %0 = stablehlo.remainder %arg0, %arg1 : tensor<32x32xf32>
    // CHECK: %[[EMPTY:[0-9]+]] = tensor.empty() : tensor<32x32xf32>
    // CHECK: %[[REM:[0-9]+]] = "ttir.remainder"(%arg0, %arg1, %[[EMPTY]]){{.*}} -> tensor<32x32xf32>
    return %0 : tensor<32x32xf32>
    // CHECK: return %[[REM]] : tensor<32x32xf32>
  }
```

@mrakitaTT wdy think?

Comment (mmanzoorTT) [reply test/ttmlir/Silicon/StableHLO/maxpool2d_op.mlir:19]: `max_pool2d` test is unsupported for now. We are having some problems with it. That's why I haven't added any filecheck till we figure out the issue.

**Review by AleksKnezevic — APPROVED** [review]

**Pull request merged**